Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Post Item/Components/Post Interaction Bar.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  File.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-06-10.
+//
+
+import SwiftUI
+
+import Foundation
+
+/**
+ View grouping post interactions--upvote, downvote, save, reply, plus post info
+ */
+struct PostInteractionBar: View {
+    // ==== TEMPORARY ====
+    @State var isPresentingAlert: Bool = false
+    // ==== END TEMPORARY ==== //
+    
+    // constants
+    let iconToTextSpacing: CGFloat = 2
+    let iconPadding: CGFloat = 4
+    let iconCorner: CGFloat = 2
+    let scoreItemWidth: CGFloat = 12
+    
+    // state fakers--these let the upvote/downvote/score/save views update instantly even if the call to the server takes longer
+    @State var dirtyVote: ScoringOperation
+    @State var dirtyScore: Int
+    @State var dirtySaved: Bool
+    @State var dirty: Bool
+    
+    // computed properties--if dirty, show dirty value, otherwise show post value
+    var displayedVote: ScoringOperation { dirty ? dirtyVote : post.myVote ?? .resetVote }
+    var displayedScore: Int { dirty ? dirtyScore : post.counts.score }
+    
+    // parameters
+    let post: APIPostView
+    let account: SavedAccount
+    let compact: Bool
+    let voteOnPost: (ScoringOperation) async -> Void
+    
+    init(post: APIPostView, account: SavedAccount, compact: Bool, voteOnPost: @escaping (ScoringOperation) async -> Void) {
+        self.post = post
+        self.account = account
+        self.compact = compact
+        self.voteOnPost = voteOnPost
+        _dirtyVote = State(initialValue: post.myVote ?? .resetVote)
+        _dirtyScore = State(initialValue: post.counts.score)
+        _dirtySaved = State(initialValue: false)
+        _dirty = State(initialValue: false)
+    }
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            HStack(spacing: 8) {
+                VoteComplex(vote: displayedVote, score: displayedScore, upvote: upvote, downvote: downvote)
+                    .padding(.trailing, 8)
+                SaveButton(saved: false)
+                    .onTapGesture {
+                        Task(priority: .userInitiated) {
+                            await savePost()
+                        }
+                        // ==== TEMPORARY ==== //
+                        isPresentingAlert = true
+                    }
+                    .alert("That feature isn't implemented yet!",
+                           isPresented: $isPresentingAlert) {
+                    }
+                // ==== END TEMPORARY ==== //
+                ReplyButton()
```

Comment:
I think the reply button should be replaced with a "share" button. Having the reply here would mean having to open the post and then focus the reply field at the bottom of the screen, which would be too busy.